ipamutils/utils.go

```diff
@@ -37,18 +37,37 @@ type NetworkToSplit struct {
 // InitNetworks initializes the broad network pool and the granular network pool
 func InitNetworks(defaultAddressPool []*NetworkToSplit) {
 	initNetworksOnce.Do(func() {
-		// error ingnored should never fail
-		PredefinedGranularNetworks, _ = splitNetworks(defaultGranularNetwork)
+		mutex.Lock()
+		// error ignored should never fail
+		PredefinedGlobalScopeDefaultNetworks, _ = splitNetworks(globalScopeDefaultNetworks)
+		mutex.Unlock()
 		if defaultAddressPool == nil {
-			defaultAddressPool = defaultBroadNetwork
+			defaultAddressPool = localScopeDefaultNetworks
 		}
 		var err error
-		if PredefinedBroadNetworks, err = splitNetworks(defaultAddressPool); err != nil {
-			logrus.WithError(err).Error("InitAddressPools failed to initialize the default address pool")
+		if PredefinedLocalScopeDefaultNetworks, err = splitNetworks(defaultAddressPool); err != nil {
+			logrus.WithError(err).Error("InitAddressPools failed to initialize the local scope default address pool")
 		}
 	})
 }
 
+// ConfigGlobalScopeDefaultNetworks initializes the global scope default network pool
+func ConfigGlobalScopeDefaultNetworks(defaultAddressPool []*NetworkToSplit) {
+	mutex.Lock()
+	if defaultAddressPool == nil {
+		defaultAddressPool = globalScopeDefaultNetworks
+	}
+	var err error
```

Comment:
why this err is defined here and not in the next line if not used later?

Reply:
while re writing , I missed it out. corrected it.

Reply:
If I move err inside if block, I am getting below error.  So moved it back up.
ipamutils/utils.go:60:2: PredefinedGlobalScopeDefaultNetworks declared and not used

Reply:
Agree.   This is one of those ... er ... charming Go scope behaviors with `:=` ...  Go will assume that `PredefinedGlobalScopeDefaultNetworks` is also a local variable in need of fresh declaration.  Go allows for partial redeclaration of variables ... except when it doesn't ... by overriding the outer declaration with an inner one.